plugins/modules/zabbix_usergroup.py

```diff
@@ -0,0 +1,597 @@
+            - Name of the user group to create, update or delete.
+        required: true
+        type: str
+        aliases: [ "user_group" ]
+    gui_access:
+        description:
+            - Frontend authentication method of the users in the group.
+            - "Possible values:"
+            - default -  use the system default authentication method;
+            - internal - use internal authentication;
+            - LDAP - use LDAP authentication;
+            - disable - disable access to the frontend.
+        required: false
+        type: str
+        default: "default"
+        choices: [ "default", "internal", "LDAP", "disable"]
+    debug_mode:
+        description:
+            - Whether debug mode is enabled or disabled.
+        required: false
+        type: str
+        default: "disabled"
+        choices: [ "disabled", "enabled" ]
+    status:
+        description:
+            - Whether the user group is enabled or disabled.
+        required: false
+        type: str
+        default: "enabled"
+        choices: [ "enabled", "disabled" ]
+    rights:
+        description:
+            - Permissions to assign to the group
+        required: false
+        type: list
+        elements: dict
+        suboptions:
+            host_group:
+                description:
+                    - Name of the host group to add permission to.
+                required: true
+                type: str
+            permission:
+                description:
+                    - Access level to the host group.
+                required: true
+                type: str
+                choices: [ "denied", "read-only", "read-write" ]
+    tag_filters:
+        description:
+            - Tag based permissions to assign to the group
+        required: false
+        type: list
+        elements: dict
+        suboptions:
+            host_group:
+                description:
+                    - Name of the host group to add permission to.
+                required: true
+                type: str
+            tag:
+                description:
+                    - Tag name.
+                required: false
+                type: str
+                default: ''
+            value:
+                description:
+                    - Tag value.
+                required: false
+                type: str
+                default: ''
+    state:
+        description:
+            - State of the user group.
+            - On C(present), it will create if user group does not exist or update the user group if the associated data is different.
+            - On C(absent) will remove a user group if it exists.
+        required: false
+        type: str
+        default: "present"
+        choices: [ "present", "absent" ]
+notes:
+    - Only Zabbix >= 4.0 is supported.
+extends_documentation_fragment:
+- community.zabbix.zabbix
+'''
+
+EXAMPLES = r'''
+# Base create user group example
+- name: Create user group
+  zabbix_usergroup:
```

Comment:
FQCN here
```suggestion
  community.zabbix.zabbix_usergroup:
```

Also this means for new modules, you could just call this `usergroup.py`

```suggestion
  community.zabbix.usergroup:
```

Reply:
@gundalow Thanks, I will prefer to stick to `zabbix_usergroup.py` as module name. Maybe this could be changed in future PR(s) along with the other modules. I know, that will break a lot of things also.
My personal concern is about breaking things if I use [Simplifying module names with the collections keyword](https://docs.ansible.com/ansible/devel/user_guide/collections_using.html#simplifying-module-names-with-the-collections-keyword) and have two or more collections with a module named `usergroup` in a play? I guess this will also be confusing or?

Beside that I will update the documentation part to use FQCN (I still have to get used to the new spelling).

Reply:
They are valid points, feel free to keep the PR name as is. Though FQCN in collection name might be worth considering.

Reply:
Which I see you already done, thanks.